lib/broccoli_sass_compiler.js

```diff
@@ -273,12 +274,23 @@ BroccoliSassCompiler.prototype.logCompilationFailure = function(details, error)
 
 BroccoliSassCompiler.prototype.compileTree = function(srcPath, files, destDir) {
   var self = this;
-
-  return files.reduce(function(promise, sassFile) {
-    return promise.then(function() {
-      return RSVP.all(self.compileSassFile(srcPath, sassFile, destDir));
-    });
-  }, RSVP.Promise.resolve());
+  //  flatten array of promises
+  var promises = [];
+  files.forEach(function(sassFile) {
+    promises = promises.concat(RSVP.all(self.compileSassFile(srcPath, sassFile, destDir)));
+  });
+  // throttle number of calls to compileSassFile
+  var numConcurrentCalls = os.cpus().length;
+  var promiseChunks = [];
+  for (var index = 0; index < promises.length; index += numConcurrentCalls) {
+    var chunk = promises.slice(index, index + numConcurrentCalls);
+    promiseChunks.push(chunk);
+  }
+ return promiseChunks.reduce(function(promise, chunk) {
+   return promise.then(function() {
+     return RSVP.all(chunk);
```

Comment:
If one of these fails, we will not wait for everything to complete before moving on. That will most likely cause issues.

I raised similar concerns [here](https://github.com/stefanpenner/broccoli-persistent-filter/pull/111) I believe @mikrostew came up with a solution that mitigates the issues. I believe you guys are on the same team, if so you should sync up.